transport/server.cc

```diff
@@ -1156,6 +1170,8 @@ process_batch_internal(service::client_state& client_state, distributed<cql3::qu
             .then([stream, batch, q_state = std::move(q_state), trace_state = query_state.get_trace_state(), version] (auto msg) {
         if (msg->move_to_shard()) {
             return process_fn_return_type(dynamic_pointer_cast<messages::result_message::bounce_to_shard>(msg));
+        } else if (msg->is_exception()) {
```

Comment:
Let's add __builtin_expect() to exception_container::is_empty() so that the compiler knows that this is an unexpected path and can optimize a little better.

Reply:
Done, added `__builtin_expect()` to the `empty` method (it was renamed from `is_empty` to `empty`).